File: php/examples/getOrdersWithRestrictedDataToken.php
<?php

require_once '../sdk/vendor/autoload.php';

use SpApi\AuthAndAuth\LWAAuthorizationCredentials;
use SpApi\Configuration;
use SpApi\Api\Orders\v0\OrdersV0Api;
use SpApi\Api\tokens\v2021_03_01\TokensApi;
use SpApi\Model\tokens\v2021_03_01\CreateRestrictedDataTokenRequest;
use SpApi\Model\tokens\v2021_03_01\RestrictedResource;
use Dotenv\Dotenv;

// Set the credentials, region and marketplace in .env file
$dotenv = Dotenv::createImmutable('../');
$dotenv->load();

// Set up LWA credentials
$lwaAuthorizationCredentials = new LWAAuthorizationCredentials([
    "clientId" => $_ENV['SP_API_CLIENT_ID'],
    "clientSecret" => $_ENV['SP_API_CLIENT_SECRET'],
    "refreshToken" => $_ENV['SP_API_REFRESH_TOKEN'],
    "endpoint" => $_ENV['SP_API_ENDPOINT']
]);

// Initialize configuration
$config = new Configuration([], $lwaAuthorizationCredentials);
$config->setHost("https://sellingpartnerapi-na.amazon.com");

try {
    // Create a restricted resource specifying the API endpoint and required data elements
    $resource = new RestrictedResource();
    $resource->setMethod('GET');
    $resource->setPath('/orders/v0/orders');
    $resource->setDataElements([
        'buyerInfo',
        'shippingAddress'
    ]);

    // Get a Restricted Data Token for accessing PII data
    $tokensApi = new TokensApi($config);
    $request = new CreateRestrictedDataTokenRequest();
    $request->setRestrictedResources([$resource]);
    $response = $tokensApi->createRestrictedDataToken($request);
    $rdtToken = $response->getRestrictedDataToken();

    // Create Orders API client
    $ordersApi = new OrdersV0Api($config);

    // Make the API call with RDT token
    $response = $ordersApi
        ->getOrders(
            ['ATVPDKIKX0DER'],  // marketplace_ids
            '2023-01-01T00:00:00Z',  // createdAfter
            null,  // createdBefore
            null,  // lastUpdatedAfter
            null,  // lastUpdatedBefore
            ['Shipped'],  // Only Unshipped or Shipped orderStatuses contain PII data
            restrictedDataToken: $rdtToken // Pass RDT token to access PII data
        );

    // Process Orders API response
    echo "Order API Response:\n";
    print_r($response);

} catch (Exception $e) {
    echo "Error: " . $e->getMessage() . "\n";
}

File: java/examples/GetOrdersWithRestrictedDataToken.java
package examples;

import com.amazon.SellingPartnerAPIAA.LWAAuthorizationCredentials;
import software.amazon.spapi.api.orders.v0.OrdersV0Api;
import software.amazon.spapi.api.tokens.v2021_03_01.TokensApi;
import software.amazon.spapi.models.orders.v0.GetOrdersResponse;
import software.amazon.spapi.models.tokens.v2021_03_01.CreateRestrictedDataTokenRequest;
import software.amazon.spapi.models.tokens.v2021_03_01.CreateRestrictedDataTokenResponse;
import software.amazon.spapi.models.tokens.v2021_03_01.RestrictedResource;


import java.util.Arrays;
import java.util.List;

public class GetOrdersWithRestrictedDataToken {

    public static void main(String[] args) {
        try {

            // Set up LWA credentials
            LWAAuthorizationCredentials lwaCredentials = LWAAuthorizationCredentials.builder()
                    .clientId("SP_API_CLIENT_ID")
                    .clientSecret("SP_API_CLIENT_SECRET")
                    .refreshToken("SP_API_REFRESH_TOKEN")
                    .endpoint("https://api.amazon.com/auth/o2/token")
                    .build();

            // Create a restricted resource specifying the API endpoint and required data elements
            RestrictedResource resource = new RestrictedResource();
            resource.setMethod(RestrictedResource.MethodEnum.GET);
            resource.setPath("/orders/v0/orders");
            resource.setDataElements(Arrays.asList("buyerInfo", "shippingAddress"));

            // Get a Restricted Data Token for accessing PII data
            TokensApi tokensApi = new TokensApi.Builder()
                    .lwaAuthorizationCredentials(lwaCredentials)
                    .endpoint("https://sellingpartnerapi-na.amazon.com")
                    .build();

            CreateRestrictedDataTokenRequest request = new CreateRestrictedDataTokenRequest();
            request.setRestrictedResources(Arrays.asList(resource));

            CreateRestrictedDataTokenResponse tokenResponse = tokensApi.createRestrictedDataToken(request);
            String rdtToken = tokenResponse.getRestrictedDataToken();

            // Create Orders API client
            OrdersV0Api ordersApi = new OrdersV0Api.Builder()
                    .lwaAuthorizationCredentials(lwaCredentials)
                    .endpoint("https://sellingpartnerapi-na.amazon.com")
                    .build();

            // Make the API call with RDT token
            List<String> marketplaceIds = Arrays.asList("ATVPDKIKX0DER");
            List<String> orderStatuses = Arrays.asList("Shipped");
            String createdAfter = "2023-01-01T00:00:00Z";


            GetOrdersResponse response = ordersApi.getOrders(
                    marketplaceIds,     // marketplaceIds
                    createdAfter,       // createdAfter
                    null,               // createdBefore
                    null,               // lastUpdatedAfter
                    null,               // lastUpdatedBefore
                    orderStatuses,      // orderStatuses
                    null,               // fulfillmentChannels
                    null,               // paymentMethods
                    null,               // buyerEmail
                    null,               // sellerOrderId
                    100,                // maxResultsPerPage
                    null,               // easyShipShipmentStatuses
                    null,               // electronicInvoiceStatuses
                    null,               // nextToken
                    null,               // amazonOrderIds
                    null,               // actualFulfillmentSupplySourceId
                    null,               // isISPU
                    null,               // storeChainStoreId
                    null,               // earliestDeliveryDateBefore
                    null,               // earliestDeliveryDateAfter
                    null,               // latestDeliveryDateBefore
                    null,               // latestDeliveryDateAfter
                    rdtToken            // restrictedDataToken
            );

            // Process Orders API response
            System.out.println("Order API Response:");
            System.out.println(response);

        } catch (Exception e) {
            System.err.println("Error: " + e.getMessage());
            e.printStackTrace();
        }
    }
}